src/charm.py

```diff
@@ -1327,12 +1329,33 @@ def _on_update_status_early_exit_checks(self, container) -> bool:
             return False
         return True
 
+    def _check_pgdata_storage_size(self) -> None:
+        """Asserts that pgdata volume has at least 10% free space and blocks charm if not."""
+        try:
+            total_size, _, free_size = shutil.disk_usage(self.pgdata_path)
+        except FileNotFoundError:
+            logger.error("pgdata folder not found in %s", self.pgdata_path)
+            return
+
+        logger.debug(
+            "pgdata free disk space: %s out of %s, ratio of %s",
+            free_size,
+            total_size,
+            free_size / total_size,
+        )
+        if free_size / total_size < 0.1:
+            self.unit.status = BlockedStatus(INSUFFICIENT_SIZE_WARNING)
+        elif self.unit.status.message == INSUFFICIENT_SIZE_WARNING:
+            self._set_active_status()
```

Comment:
We should most probably don't change the status in `_set_active_status()` if the message is `INSUFFICIENT_SIZE_WARNING` and then reset the status here before running `_set_active_status()`.

This is because if another hook runs `_set_active_status()` the status will be overwritten unite the next update status. @marceloneppel what do you think?

Reply:
I'm not sure I understand your point. Could you elaborate? My motivation for this check is to resolve the blocked status in case the storage space got released (to >10%)

Reply:
the check for `INSUFFICIENT_SIZE_WARNING` is just to make sure that, if the charm was blocked due to insufficient size but the user has released enough space, then resolve. Maybe there's a better way?

Reply:
I think it's a good idea, @dragomirp.

@lucasgameiroborges, the idea is to just set something else in the status at this point, like the Active Status, and put a condition in the `_set_active_status` method not to change the status to Active if the current status is `INSUFFICIENT_SIZE_WARNING`. This way, we avoid some other places overring this status incorrectly (maybe [here](https://github.com/canonical/postgresql-k8s-operator/blob/a3f363a383da6969390ab3579343f1390aa43acd/src/charm.py#L940)).

I see that this will need to be handled correctly later by improving how we manage the unit's statuses across the entire charm code.

Reply:
I get the reasoning for including a check inside `_set_active_status` so the blocked status doesn't get overwritten in other places, as for `just set something else in the status at this point, like the Active Status` do you mean using `self.unit.status = ActiveStatus()` directly? why?

Reply:
The elif should be something like:
```
        elif self.unit.status.message == INSUFFICIENT_SIZE_WARNING:
            self.unit.status = ActiveStatus()
            self._set_active_status()
```
if `_set_active_status()` preserves `INSUFFICIENT_SIZE_WARNING` it should be unset before calling it. Otherwise it won't unset.

Reply:
Q: are we moving the right way here? 
It smells like Status/Message manipulation is going outside of our control.

Reply:
Oh! Now I see the problem! Yes of course, thanks @dragomirp and @marceloneppel !

@taurus-forever : I agree that status management is a bit all over the place. AFAIK this is a problem in other charms too. Interesting discussion here: https://chat.canonical.com/canonical/pl/ncbsrryk378jux5r8cxjjngzna